OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs

```diff
@@ -294,9 +294,9 @@ bool HandleRightClick(ProductionItem item, ProductionIcon icon, int handleCount)
 
 			Game.Sound.Play(SoundType.UI, TabClick);
 
-			if (item.Paused || item.Done || item.TotalCost == item.RemainingCost)
+			if (CurrentQueue.Info.DisallowOnHold || item.Paused || item.Done || item.TotalCost == item.RemainingCost)
 			{
-				// Instant cancel of things we have not started yet and things that are finished
+				// Instantly cancel things we have not started yet and things that are finished or if the queue doesn't allow on hold
```

Comment:
IMO the following would be a bit clearer: `// Instantly cancel items that haven't started, have finished, or if the queue doesn't support pausing.`